Source/Stationeers.Addons/Modules/Workshop/WorkshopManagerPatch.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Assets.Scripts;
+using Assets.Scripts.Steam;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Stationeers.Addons.Modules.Workshop
+{
+    public class WorkshopManagerPatch
+    {
+        // Why does System.IO not have a directory copy method?
+        // Stolen from here: https://stackoverflow.com/questions/1974019/folder-copy-in-c-sharp
+        private static void copyDirectory(string strSource, string strDestination)
+        {
+            if (!Directory.Exists(strDestination))
+            {
+                Directory.CreateDirectory(strDestination);
+            }
+
+            DirectoryInfo dirInfo = new DirectoryInfo(strSource);
+            FileInfo[] files = dirInfo.GetFiles();
+            foreach (FileInfo tempfile in files)
+            {
+                tempfile.CopyTo(Path.Combine(strDestination, tempfile.Name));
+            }
+
+            DirectoryInfo[] directories = dirInfo.GetDirectories();
+            foreach (DirectoryInfo tempdir in directories)
+            {
+                copyDirectory(Path.Combine(strSource, tempdir.Name), Path.Combine(strDestination, tempdir.Name));
+            }
+
+        }
+
+        public static void PublishWorkshopPrefix(WorkshopManager __instance, ref WorkShopItemDetail ItemDetail, ref string changeNote, out string __state)
+        {
+            string origItemContentPath = ItemDetail.Path;
+            string tempItemContentPath = origItemContentPath + "_temp";
+            __state = "";
+
+            DirectoryInfo tempItemContentDir = Directory.CreateDirectory(tempItemContentPath);
+
+            // Copy files to upload to temp directory, if they satisfy upload whitelist.
+            foreach (string itemFilePath in Directory.GetFiles(origItemContentPath))
+            {
+                bool validFile = false;
+                string fileName = new FileInfo(itemFilePath).Name;
+
+                // Ugly fall-through regex party.
+                if (new Regex(@".*\.cs$").IsMatch(fileName))
```

Comment:
We could think of an table of regex strings for this.
Something like:
```CS
private static readonly Regex[] ValidFileNames = {
    new Regex(@".*\.cs$"),
    new Regex(@".*\.xml$")
    // ...
};
```
(or just strings if it does not work, didn't have the time to test it ;) )
And then use LINQ:
```CS
var validFile = ValidFileNames.Any(regex => regex.IsMatch(fileName));
```